File: app/src/main/java/com/paxus/pay/poslinkui/demo/MainActivity.java
package com.paxus.pay.poslinkui.demo;

import android.content.Context;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.util.AttributeSet;
import android.view.View;
import android.widget.CheckedTextView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

import com.paxus.pay.poslinkui.demo.settings.EntryActionAndCategoryFilterFragment;

/**
 * Created by Yanina.Yang on 5/13/2022.
 * <p>
 * Add a log on Android HOME for easy install&uninstall
 */
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        try {
            String appVer = getPackageManager().getPackageInfo(getPackageName(), 0).versionName;
            ((TextView) findViewById(R.id.version)).setText("Version: " + appVer);
        } catch (PackageManager.NameNotFoundException e) {
            e.printStackTrace();
        }

        findViewById(R.id.customize_filter).setOnClickListener(view -> {
            getSupportFragmentManager().executePendingTransactions();
            getSupportFragmentManager().beginTransaction()
                    .addToBackStack(EntryActionAndCategoryFilterFragment.class.getSimpleName())
                    .setCustomAnimations(R.anim.anim_enter_from_bottom, R.anim.anim_exit_to_bottom)
                    .replace(R.id.activity_main_fragment_container, new EntryActionAndCategoryFilterFragment()).commit();
            ((CheckedTextView)view).setChecked(true);
        });

        getSupportFragmentManager().registerFragmentLifecycleCallbacks(new FragmentManager.FragmentLifecycleCallbacks() {
            @Override public void onFragmentDetached(@NonNull FragmentManager fm, @NonNull Fragment f) {
                super.onFragmentDetached(fm, f);
                ((CheckedTextView)findViewById(R.id.customize_filter)).setChecked(false);
            }
        }, false);
    }

    @Override
    protected void onStop() {
        super.onStop();
        finish();
    }
}

File: app/src/main/java/com/paxus/pay/poslinkui/demo/MainActivity.kt
package com.paxus.pay.poslinkui.demo

import android.content.pm.PackageManager
import android.os.Bundle
import android.view.View
import android.widget.CheckedTextView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import com.paxus.pay.poslinkui.demo.settings.EntryActionAndCategoryFilterFragment
import com.paxus.pay.poslinkui.demo.utils.AnimationSupport
import com.paxus.pay.poslinkui.demo.utils.Logger
import com.paxus.pay.poslinkui.demo.utils.currentAnimationPolicy
import dagger.hilt.android.AndroidEntryPoint

/**
 * Launcher activity for the POSLink UI demo: shows build version, opens the entry filter UI from the
 * customize row, and registers fragment callbacks so that row state stays consistent.
 *
 * Exposed as the HOME intent target to simplify install/uninstall during development.
 */
@AndroidEntryPoint
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        supportActionBar?.hide()

        try {
            val appVer = packageManager.getPackageInfo(packageName, 0).versionName
            (findViewById<View?>(R.id.version) as TextView).text = "Version: $appVer"
        } catch (e: PackageManager.NameNotFoundException) {
            Logger.e(e, "Could not resolve package info for version label")
        }

        findViewById<View?>(R.id.customize_filter).setOnClickListener { view: View? ->
            supportFragmentManager.executePendingTransactions()
            val fragmentManager = supportFragmentManager
            val backStackName = EntryActionAndCategoryFilterFragment::class.java.simpleName
            val baseTransaction = fragmentManager.beginTransaction()
                .setReorderingAllowed(true)
                .addToBackStack(backStackName)
            val withAnimations = AnimationSupport.applyFragmentTransition(
                transaction = baseTransaction,
                pointId = "A1",
                policy = currentAnimationPolicy,
                standardEnter = R.anim.anim_enter_from_bottom,
                standardExit = R.anim.anim_exit_to_bottom,
            )
            withAnimations.replace(
                R.id.activity_main_fragment_container,
                EntryActionAndCategoryFilterFragment(),
            ).commit()
            (view as CheckedTextView).isChecked = true
        }

        supportFragmentManager.registerFragmentLifecycleCallbacks(object :
            FragmentManager.FragmentLifecycleCallbacks() {
            override fun onFragmentDetached(fm: FragmentManager, f: Fragment) {
                super.onFragmentDetached(fm, f)
                (findViewById<View>(R.id.customize_filter) as? CheckedTextView)?.isChecked = false
            }
        }, false)
    }

    override fun onStop() {
        super.onStop()
        finish()
    }
}
